graphistry/plugins/igraph.py

```diff
@@ -209,3 +214,164 @@ def to_igraph(self,
     node_attrs = [x for x in node_attrs if x != g._node]
     nodes_df = g._nodes[[g._node] + node_attrs]
     return igraph.Graph.DataFrame(edges_df, directed=directed, vertices=nodes_df)
+
+
+compute_algs = [
+    'authority_score',
+    'betweenness',
+    'bibcoupling',
+    #'bipartite_projection',
+    'harmonic_centrality',
+    'closeness',
+    'clusters',
+    'cocitation',
+    'community_edge_betweenness',
+    'community_fastgreedy',
+    'community_infomap',
+    'community_label_propagation',
+    #'community_leading_eigenvector_naive',  # in docs but not in code?
+    'community_leading_eigenvector',
+    'community_leiden',
+    'community_multilevel',
+    'community_spinglass',
+    'community_walktrap',
+    'constraint',
+    'gomory_hu_tree',
+    'hub_score',
+    'eccentricity',
+    'eigenvector_centrality',
+    'k_core',
+    #'modularity',
+    'pagerank',
+    'spanning_tree'
+]
+
+def compute_igraph(
+    self: Plottable, alg: str, alg_as: Optional[str] = None, directed: Optional[bool] = None, params: dict = {}
+) -> Plottable:
+    """Enrich or replace graph using igraph methods
+
+    :param alg: Name of an igraph.Graph method like `pagerank`
+    :type alg: str
+
+    :param alg_as: For algorithms that compute values for nodes, which attribute to write to. If None, use the algorithm's name. (default None)
```

Comment:
Docstring is not clear in my first reading of it.

Reply:
great, changing output cols to `out_col`, `x_out_col`, and `y_out_col`

Reply:
and adding examples